src/components/Map.js

```diff
@@ -20,29 +20,51 @@ const Map = ({
   setMapProperties,
   center,
 }) => {
-  useEffect(() => {
-    fetchNearbyPlaces(
-      mapProperties.center.lat,
-      mapProperties.center.lng
-    ).then((res) => setNearbyPlaces(res));
-  }, [mapProperties.center.lat, mapProperties.center.lng, setNearbyPlaces]);
 
+  let [pendingPromise, setPendingPromise] = useState(false);
+
+  useEffect(() => {
+    if (!pendingPromise) {
+      fetchNearbyPlaces(
+        mapProperties.center.lat,
+        mapProperties.center.lng,
+        setPendingPromise
+      ).then((res) => {setNearbyPlaces(res); setPendingPromise(false);});
+    }
+  }, []);
 
   // only runs if maps object is populated
-  if (!isObjEmpty(mapsObj)) {
+  if (!isObjEmpty(mapsObj) && !pendingPromise) {
     const { map } = mapsObj;
-    map.addListener("center_changed", function () {
+    // map.addListener("center_changed", function () {
+    //   const panCoords = map.getCenter();
+    //   const panLat = panCoords.lat();
+    //   const panLng = panCoords.lng();
+    //   setUserHasPanned(true);
+    //   if(!pendingPromise){
+    //     handleCenterChanged(panLat, panLng);
+    //   }
+    // })
+    map.addListener("dragend", function (event) {
```

Comment:
`dragend` is the best event for this. curious if you tried `idle` and whether you noticed a difference?

Reply:
I just tried `idle` and it made no difference